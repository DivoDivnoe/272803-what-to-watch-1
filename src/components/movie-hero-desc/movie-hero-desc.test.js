import React from 'react';
import renderer from 'react-test-renderer';
import MovieHeroDesc from './movie-hero-desc.jsx';
import {BrowserRouter} from 'react-router-dom';

const mock = {
  movie: {
    id: 3,
    name: `Major Payne`,
    description: `very interesting film`,
    director: `Steven Spielberg`,
    starring: [`Andrey Ivanov`, `Sergey Rubets`],
    rating: 10,
    scoresCount: 100000,
    backgroundImage: ``,
    backgroundColor: `cyan`,
    released: 2019,
    posterImage: ``,
    genre: `Comedy`,
    previewVideoLink: ``,
    previewImage: ``,
    videoLink: ``,
    runTime: 600,
    isFavorite: false,
  },
  reviewsLinkRequired: true,
  isInList: true,
  onSwitchPlayer: jest.fn(),
  onSetToFavorites: jest.fn(),
};

describe(`MovieHeroDesc component`, () => {
  const {
    movie,
    reviewsLinkRequired,
    isInList,
    onSwitchPlayer,
    onSetToFavorites,
  } = mock;

  it(`renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <MovieHeroDesc
            movie={movie}
            reviewsLinkRequired={reviewsLinkRequired}
            isInList={isInList}
            onSwitchPlayer={onSwitchPlayer}
            onSetToFavorites={onSetToFavorites}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
