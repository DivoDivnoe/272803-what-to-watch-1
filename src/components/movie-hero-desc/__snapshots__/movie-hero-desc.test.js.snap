// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieHeroDesc component renders correctly 1`] = `
<div
  className="movie-card__desc"
>
  <h2
    className="movie-card__title"
  >
    Major Payne
  </h2>
  <p
    className="movie-card__meta"
  >
    <span
      className="movie-card__genre"
    >
      Comedy
    </span>
    <span
      className="movie-card__year"
    >
      2019
    </span>
  </p>
  <div
    className="movie-card__buttons"
  >
    <button
      className="btn btn--play movie-card__button"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <button
      className="btn btn--list movie-card__button"
      disabled={true}
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        height="14"
        viewBox="0 0 18 14"
        width="18"
      >
        <use
          xlinkHref="#in-list"
        />
      </svg>
      <span>
        My list
      </span>
    </button>
    <a
      className="btn movie-card__button"
      href="/"
      onClick={[Function]}
    >
      Add review
    </a>
  </div>
</div>
`;
